dashboard/app/jobs/cap/lockout_job.rb

```diff
@@ -1,14 +1,36 @@
 module CAP
   # Attempts to transition the User to the locked-out state if they
-  # should no longer be able to access Code.org due to their State's 
+  # should no longer be able to access Code.org due to their State's
   # Child Account Policy.
   class LockoutJob < ApplicationJob
+    MAX_RESCHEDULING_ATTEMPTS = 1
+
     rescue_from StandardError, with: :report_exception
 
+    # Schedules the user lockout based on their estimated lockout date.
+    # @param user [User] the student account
+    # @param reschedules [Integer] the number of times the job can be rescheduled
+    # @return [CAP::LockoutJob, nil] the scheduled lockout job on success, nil otherwise
+    def self.schedule_for(user, reschedules: MAX_RESCHEDULING_ATTEMPTS)
+      estimated_lockout_date = Policies::ChildAccount.lockout_date(user)
+      return unless estimated_lockout_date
+
+      # Schedules the lockout job to run one minute later
+      # to ensure it does not perform before the estimated lockout time.
+      scheduled_lockout_date = estimated_lockout_date.since(1.minute)
+      set(wait_until: scheduled_lockout_date).perform_later(user_id: user.id, reschedules: reschedules)
+    end
+
     # @param user_id [Integer] the student account ID
-    def perform(user_id:)
+    # @param reschedules [Integer] the number of times the job can be rescheduled due to a lockout failure
+    def perform(user_id:, reschedules: 0)
       user = User.find(user_id)
-      Services::ChildAccount::LockoutHandler.call(user: user)
+
+      user_is_locked_out = Services::ChildAccount::LockoutHandler.call(user: user)
+
+      # If the user was not locked out, attempt to reschedule the lockout job
+      # in case the reason is a re-estimation of the user's lockout date.
+      self.class.schedule_for(user, reschedules: reschedules.pred) unless user_is_locked_out || reschedules <= 0
```

Comment:
No, because `Policies::ChildAccount.lockout_date` will return `nil` if the student is CAP-compliant or already locked out.

Reply:
`self.class.schedule_for` will be called, but the job will not be scheduled because `Policies::ChildAccount.lockout_date` returns `nil` if the student is either CAP-compliant or already locked out.

https://github.com/code-dot-org/code-dot-org/blob/1469cc5976a972ac733d2991f2b792ed61cb3ae4/dashboard/app/jobs/cap/lockout_job.rb#L15-L16

Reply:
Oh yeah and we already had a long discussion about that in this PR, sorry for forgetting!